### PR TITLE
BVH Animation to SEA3D Character and SkeletonUtils

### PR DESCRIPTION
**Pull request by sunag**
https://rawgit.com/sunag/three.js/dev-sea3d-bvh/examples/index.html?q=sea3d#webgl_loader_sea3d_bvh

### CODE REVIEW

**Comment by Mugen87** [comment]
No offense against `SEA3D`^^. But since `glTF` is the recommended format of the project it would be better if the examples use `GLTFLoader`. Do you mind switching to `glTF`?

**Comment by sunag** [comment]
> No offense against SEA3D^^. But since glTF is the recommended 3D format of the project it would be better if the examples use GLTFLoader. Do you mind switching to glTF?

Maybe this will never happen I see SEA3D evolving independently of others formats. SEA3D has more features and a better file structure in my opinion.

**Comment by looeee** [comment]
> SEA3D has more features and a better file structure in my opinion.

That may be the case, but the most important thing for 3D format is the level of adoption. In that regard glTF is the clear winner here and I have to agree with @Mugen87 on this.

**Comment by sunag** [comment]
> That may be the case, but the most important thing for 3D format is the level of adoption. In that regard glTF is the clear winner here and I have to agree with @Mugen87 on this.

Sorry but for me quality is more important than fame for anything. But for a exchange format I could recommend OBJ, FBX or glTF.

**Comment by donmccurdy** [comment]
I don't necessarily think we need to pick a single format for examples... I might have more concerns if this were listed as a general example, or if the SEA3D loader weren't well maintained, but it is listed under the SEA3D example category and @sunag is clearly actively maintaining things (thanks!).

We definitely do need more, and better, glTF examples to help new users. I've been meaning to work on that — there are some good glTF examples with CC licenses on Sketchfab — but I haven't found the time. That can happen without removing or preventing examples of other formats I think, though.

**Comment by looeee** [comment]
> quality is more important than fame for anything

It's not at all about fame. It's about usability. If there's a format that is basically only be used by one person then it really doesn't matter how good it is, it's not really useful. 

@Sunag this is not to put down your work - I don't doubt that SEA3D format is as good as you say it is, but we need to consider how useful these examples are for three.js consumers, and also how likely they are to be found from a search engine. For these reasons in my opinion we should stick to glTF, FBX and OBJ for new examples.

**Comment by sunag** [comment]
@looeee SEA3D has users who found in SEA3D what they do not find in others formats and this goes for any other format. This goes according to the need of each one, so some will always prefer glTF and others SEA3D. I do not think that streets should are made only for a model of car.

File: examples/js/utils/SkeletonUtils.js [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+					}
+
+					globalMatrix.copyPosition( relativeMatrix );
+
+				}
+
+				if ( bone.parent && bone.parent.isBone ) {
+
+					bone.matrix.getInverse( bone.parent.matrixWorld );
+					bone.matrix.multiply( globalMatrix );
```
[the remark this answers is not in the document]
Comment (sunag) [reply examples/js/utils/SkeletonUtils.js]: Hmm I am not quite right about this but I think who had I fixed these issue with this code:
https://github.com/sunag/three.js/blob/a98e955043eeec5830c7b865dfd63696ecf8a075/examples/js/utils/SkeletonUtils.js#L130

A skeleton retargeting must ignore the scale completely, position too with exception of `hip` unless if the character is the Dhalsim of Street Fighter :grin:

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**